it/controllers/pensions/WorkplaceAmountControllerISpec.scala

```diff
@@ -0,0 +1,430 @@
+          textOnPageCheck(user.specificExpectedResults.get.expectedYouCanFindThisOut, paragraphSelector(4))
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck(amountInputName, inputSelector, amountInvalidFormat)
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(workplacePensionAmount(taxYearEOY), formSelector)
+          errorSummaryCheck(invalidFormatErrorText, expectedErrorHref)
+          errorAboveElementCheck(invalidFormatErrorText)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "return an error when form is submitted with input over maximum allowed value" which {
+
+          val amountOverMaximum = "100,000,000,000"
+          val overMaximumForm: Map[String, String] = Map(AmountForm.amount -> amountOverMaximum)
+
+          lazy val result: WSResponse = {
+            dropPensionsDB()
+            val pensionsViewModel = aPaymentsIntoPensionViewModel.copy(
+              workplacePensionPaymentsQuestion = Some(true), totalWorkplacePensionPayments = None)
+            insertCyaData(pensionsUsersData(isPrior = false, aPensionsCYAModel.copy(paymentsIntoPension = pensionsViewModel)), aUserRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlPost(fullUrl(workplacePensionAmount(taxYearEOY)), body = overMaximumForm, welsh = user.isWelsh,
+              follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          "has the correct status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          textOnPageCheck(expectedParagraph, paragraphSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedBullet1, bulletListSelector(1))
+          textOnPageCheck(user.specificExpectedResults.get.expectedBullet2, bulletListSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedYouCanFindThisOut, paragraphSelector(4))
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck(amountInputName, inputSelector, amountOverMaximum)
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(workplacePensionAmount(taxYearEOY), formSelector)
+          errorSummaryCheck(maxAmountErrorText, expectedErrorHref)
+          errorAboveElementCheck(maxAmountErrorText)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "redirect to the correct page when a valid amount is submitted and update the session amount" which {
```

Comment:
Also can move this redirect test out of the foreach loop

Reply:
Here we can test when the user is agent, and when the user is not agent, that's the reasoning for leaving in the for loop.